pkg/cache/clusterqueue_test.go

```diff
@@ -620,3 +621,207 @@ func TestClusterQueueUpdateWithAdmissionCheck(t *testing.T) {
+					},
+				},
+			},
+		},
+		"usage below nominal": {
+			cq: ClusterQueue{
+				ResourceStats: ResourceStats{
+					corev1.ResourceCPU: {
+						Nominal:  2_000,
+						Lendable: 2_000,
+						Usage:    1_000,
+					},
+					"example.com/gpu": {
+						Nominal:  5,
+						Lendable: 5,
+						Usage:    2,
+					},
+				},
+				Cohort: &Cohort{
+					ResourceStats: ResourceStats{
+						corev1.ResourceCPU: {
+							Nominal:  10_000,
+							Lendable: 10_000,
+							Usage:    2_000,
+						},
+						"example.com/gpu": {
+							Nominal:  10,
+							Lendable: 10,
+							Usage:    6,
+						},
+					},
+				},
+			},
+			wantDRName: corev1.ResourceCPU, // due to alphabetical order.
+		},
+		"usage above nominal": {
+			cq: ClusterQueue{
+				ResourceStats: ResourceStats{
+					corev1.ResourceCPU: {
+						Nominal:  2_000,
+						Lendable: 2_000,
+						Usage:    3_000,
+					},
+					"example.com/gpu": {
+						Nominal:  5,
+						Lendable: 5,
+						Usage:    7,
+					},
+				},
+				Cohort: &Cohort{
+					ResourceStats: ResourceStats{
+						corev1.ResourceCPU: {
+							Nominal:  10_000,
+							Lendable: 10_000,
+							Usage:    10_000,
+						},
+						"example.com/gpu": {
+							Nominal:  10,
+							Lendable: 10,
+							Usage:    10,
+						},
+					},
+				},
+			},
+			wantDRName:  "example.com/gpu",
+			wantDRValue: 20, // (7-5)/10
+		},
+		"one resource above nominal": {
+			cq: ClusterQueue{
+				ResourceStats: ResourceStats{
+					corev1.ResourceCPU: {
+						Nominal:  2_000,
+						Lendable: 2_000,
+						Usage:    3_000,
+					},
+					"example.com/gpu": {
+						Nominal:  5,
+						Lendable: 5,
+						Usage:    3,
+					},
+				},
+				Cohort: &Cohort{
+					ResourceStats: ResourceStats{
+						corev1.ResourceCPU: {
+							Nominal:  10_000,
+							Lendable: 10_000,
+							Usage:    10_000,
+						},
+						"example.com/gpu": {
+							Nominal:  10,
+							Lendable: 10,
+							Usage:    10,
+						},
+					},
+				},
+			},
+			wantDRName:  corev1.ResourceCPU,
+			wantDRValue: 10, // (3-2)/10
+		},
+		"usage with workload above nominal": {
+			cq: ClusterQueue{
+				ResourceStats: ResourceStats{
+					corev1.ResourceCPU: {
+						Nominal:  2_000,
+						Lendable: 2_000,
+						Usage:    1_000,
+					},
+					"example.com/gpu": {
+						Nominal:  5,
+						Lendable: 5,
+						Usage:    2,
+					},
+				},
+				Cohort: &Cohort{
+					ResourceStats: ResourceStats{
+						corev1.ResourceCPU: {
+							Nominal:  10_000,
+							Lendable: 10_000,
+							Usage:    2_000,
+						},
+						"example.com/gpu": {
+							Nominal:  10,
+							Lendable: 10,
+							Usage:    6,
+						},
+					},
+				},
+			},
+			workload: &workload.Info{
+				TotalRequests: []workload.PodSetResources{{
+					Requests: workload.Requests{
+						corev1.ResourceCPU: 4_000,
+						"example.com/gpu":  4,
+					},
+				}},
+			},
+			wantDRName:  corev1.ResourceCPU,
+			wantDRValue: 30, // (1+4-2)/10
+		},
+		"A resource with zero nominal": {
```

Comment:
I see this triggers division by 0 scenario, but maybe a slightly more realistic scenario is to set `lendingLimit: 0` and `nominal>0` (as mentioned in the other comment)?